File: android/app/src/main/kotlin/dev/odeon/android/ui/serie/TelaDaSerie.kt
package dev.odeon.android.ui.serie

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.aspectRatio
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import coil3.compose.AsyncImage
import dev.odeon.android.ui.Cores
import dev.odeon.android.ui.Recado
import dev.odeon.android.ui.RotuloDeSecao
import dev.odeon.android.ui.Tipo

/// A ficha de uma série, no celular — o mesmo desenho da TV, na largura da mão.
///
/// ## ⚠️ O que muda do desenho da sala, e por quê
///
/// | | |
/// |---|---|
/// | o pano de fundo é **um bloco 16:9 no topo**, não a tela toda | num celular em pé, arte de fundo atrás de texto vira textura: a coluna é estreita e o texto atravessa a imagem inteira |
/// | o botão principal é **largura cheia** | é o alvo mais tocado da tela, e a régua da casa é 48dp de altura mínima |
/// | as temporadas rolam **na horizontal** | vertical empurraria a primeira ação pra fora da dobra numa série de 5 |
///
/// O resto é igual, de propósito: mesmas palavras, mesma ordem, mesma decisão
/// sobre o que aparece. Ver `docs/SERIES.md` e `ModeloDaSerie`.
@Composable
fun TelaDaSerie(
    modelo: ModeloDaSerie,
    aoAbrirTemporada: (numero: Int) -> Unit,
    aoTocar: (episodioId: String, em: Double) -> Unit,
    aoVoltar: () -> Unit,
    modifier: Modifier = Modifier,
) {
    val estado by modelo.estado.collectAsStateWithLifecycle()

    when {
        estado.erro != null -> Recado(
            titulo = "a série não abriu",
            detalhe = estado.erro,
            aoTentar = modelo::carregar,
            aoVoltar = aoVoltar,
            modifier = modifier,
        )

        estado.carregando -> EsqueletoDaSerie(modifier)

        estado.vazio -> Recado(
            titulo = "esta série está sem episódios",
            detalhe = "o acervo tem a série, mas nenhum arquivo casou com ela",
            aoTentar = null,
            aoVoltar = aoVoltar,
            modifier = modifier,
        )

        else -> LazyColumn(
            modifier.fillMaxSize().background(Cores.fundo),
            contentPadding = PaddingValues(bottom = 28.dp),
        ) {
            item {
                Box(Modifier.fillMaxWidth().aspectRatio(16f / 9f)) {
                    modelo.arte(estado.panoDeFundo)?.let {
                        AsyncImage(
                            model = it,
                            contentDescription = null,
                            contentScale = ContentScale.Crop,
                            modifier = Modifier.fillMaxSize(),
                        )
                    }
                    /// ⚠️ O véu vertical existe pro título **encostar** na arte
                    /// sem flutuar sobre ela: o letreiro nasce na base do bloco,
                    /// e sem o degradê ele cairia sobre um céu claro.
                    Box(
                        Modifier.fillMaxSize().background(
                            Brush.verticalGradient(
                                0.35f to Color.Transparent,
                                1f to Cores.fundo,
                            ),
                        ),
                    )

                    /// ## ⚠️ E um véu **no topo**, pro «voltar» — 19/08/2026
                    ///
                    /// Medido no emulador, n'`A Feiticeira`: o dourado da casa
                    /// (#e0b062, luz 181) sobre o céu claro do backdrop (luz 204
                    /// onde o texto cai) dá **1,33:1**. O mínimo legível pra
                    /// texto é 4,5:1 — ou seja, o botão estava na tela e não dava
                    /// pra ler.
                    ///
                    /// O véu de baixo já existia pelo mesmo motivo, e cobria o
                    /// **letreiro**; ninguém tinha olhado pro canto de cima.
                    ///
                    /// ⚠️ **Duas tentativas antes desta, e as duas medidas.**
                    /// Com 22% do bloco o contraste foi de 1,33 pra 1,47 — nada,
                    /// porque o botão não mora no topo do bloco e sim **abaixo da
                    /// barra de status** (`bounds=[32,100][146,153]`), onde um
                    /// degradê curto já virou transparente. Subindo pra 88% de
                    /// preto em 42% do bloco deu 3,43:1 — passa o limiar de texto
                    /// **grande** (3:1) e não o de texto normal (4,5:1), e ainda
                    /// custava um canto de arte quase preto.
                    ///
                    /// Então o véu voltou a ser discreto e quem ficou legível foi
                    /// o **botão**, com fundo próprio logo abaixo. Escurecer a
                    /// obra inteira pra salvar 114 pixels de texto era pagar caro
                    /// no lugar errado.
                    Box(
                        Modifier.fillMaxSize().background(
                            Brush.verticalGradient(
                                0f to Color.Black.copy(alpha = 0.45f),
                                0.30f to Color.Transparent,
                            ),
                        ),
                    )
                    /// ⚠️ **`statusBarsPadding` aqui, e não no `Box`** — visto no
                    /// emulador em 18/08/2026: o bloco de arte é borda a borda de
                    /// propósito (ele sobe até debaixo do relógio, como a ficha),
                    /// e sem isto o «voltar» nascia **em cima do relógio**.
                    ///
                    /// O padding é do botão, então a arte continua sangrando e só
                    /// o alvo de toque desce.
                    /// ⚠️ **O botão carrega o próprio fundo**, e é isso que o faz
                    /// legível sobre qualquer arte: medido n'`A Feiticeira`, o
                    /// dourado sobre o céu claro do backdrop dava **1,33:1** — o
                    /// mínimo pra texto é 4,5:1. Não é caso raro: metade das
                    /// séries deste acervo tem backdrop claro.
                    ///
                    /// A pílula é a mesma forma que o resto do app usa pra pousar
                    /// cromo sobre imagem, e custa só a área do próprio rótulo.
                    TextButton(
                        onClick = aoVoltar,
                        modifier = Modifier
                            .align(Alignment.TopStart)
                            .statusBarsPadding()
                            .padding(start = 8.dp, top = 4.dp)
                            .background(
                                /// ⚠️ 0,70 e não 0,55: medido, a 0,55 o contraste
                                /// parou em **4,08:1**, e o mínimo pra texto é
                                /// 4,5. Meia dúzia de pontos de alfa custa nada e
                                /// é a diferença entre passar e quase passar.
                                Color.Black.copy(alpha = 0.70f),
                                RoundedCornerShape(50),
                            ),
                    ) {
                        Text("‹ voltar", color = Cores.destaque)
                    }
                }
            }

            item {
                Column(Modifier.padding(horizontal = 20.dp)) {
                    Text(
                        text = estado.titulo,
                        style = MaterialTheme.typography.headlineSmall,
                        color = Cores.texto,
                    )
                    Spacer(Modifier.height(8.dp))
                    Text(
                        text = contagem(
                            estado.temporadas.size,
                            estado.quantosEpisodios,
                            estado.quantosVistos,
                            estado.ano,
                        ),
                        style = Tipo.rotulo,
                        color = Cores.textoApagado,
                    )
                    /// ⚠️ A sinopse da série — 115 das 120 têm. Quem não tem não
                    /// ganha parágrafo nenhum (§24).
                    estado.sinopse?.let { sinopse ->
                        Spacer(Modifier.height(14.dp))
                        Text(
                            text = sinopse,
                            style = MaterialTheme.typography.bodyMedium,
                            color = Cores.textoApagado,
                            maxLines = 4,
                            overflow = TextOverflow.Ellipsis,
                        )
                    }

                    Spacer(Modifier.height(20.dp))

                    estado.ondeParou?.let { onde ->
                        val cod = onde.episodio.codigo?.let { "$it · " } ?: ""
                        Box(
                            Modifier
                                .fillMaxWidth()
                                .heightIn(min = 52.dp)
                                .clip(RoundedCornerShape(26.dp))
                                .background(Cores.destaqueQuente)
                                .clickable {
                                    aoTocar(onde.episodio.id, onde.episodio.ondeParou ?: 0.0)
                                }
                                .padding(horizontal = 22.dp, vertical = 14.dp),
                            contentAlignment = Alignment.Center,
                        ) {
                            Text(
                                text = (if (onde.comecado) "▸  continuar  " else "▸  começar  ") +
                                    "$cod${onde.episodio.title}",
                                style = MaterialTheme.typography.titleSmall,
                                color = Cores.fundoAfundado,
                                maxLines = 1,
                                overflow = TextOverflow.Ellipsis,
                            )
                        }
                        /// §24: «do começo» só havendo meio.
                        if (onde.comecado) {
                            TextButton(onClick = { aoTocar(onde.episodio.id, 0.0) }) {
                                Text("do começo", color = Cores.destaque)
                            }
                        }
                    }

                    RotuloDeSecao("temporadas", numero = estado.temporadas.size)
                }
            }

            item {
                LazyRow(
                    contentPadding = PaddingValues(horizontal = 20.dp),
                    horizontalArrangement = Arrangement.spacedBy(14.dp),
                ) {
                    items(estado.temporadas, key = { it.numero }) { t ->
                        CartaoDaTemporada(
                            temporada = t,
                            arte = modelo.arte(t.arte),
                            aoTocar = { aoAbrirTemporada(t.numero) },
                        )
                    }
                }
            }
        }
    }
}

@Composable
private fun CartaoDaTemporada(
    temporada: TemporadaDaSerie,
    arte: String?,
    aoTocar: () -> Unit,
) {
    /// ## ⚠️ O cartão de temporada virou **retrato** — 18/08/2026
    ///
    /// Ele era 16:9 porque a única imagem que havia era o `still` do primeiro
    /// episódio. Desde hoje o servidor manda o **pôster da temporada** do TMDB
    /// (461 de 473), e pôster é 2:3 — enfiá-lo num quadro deitado cortava a arte
    /// inteira: a `Temporada 1` do Arcane virou um olho.
    ///
    /// É a mesma lição do episódio, ao contrário: **a moldura segue a imagem**,
    /// e não o contrário.
    ///
    /// ⚠️ As 12 sem pôster caem no `still` do primeiro episódio, e aí é um 16:9
    /// dentro de um 2:3 — cortado nas laterais, que é o corte menos destrutivo
    /// dos dois.
    Column(
        Modifier
            .width(150.dp)
            .clip(RoundedCornerShape(12.dp))
            .clickable(onClick = aoTocar),
    ) {
        Box(Modifier.fillMaxWidth().aspectRatio(2f / 3f).clip(RoundedCornerShape(12.dp))) {
            if (arte != null) {
                AsyncImage(
                    model = arte,
                    contentDescription = null,
                    contentScale = ContentScale.Crop,
                    modifier = Modifier.fillMaxSize(),
                )
            } else {
                Box(Modifier.fillMaxSize().background(Cores.fundoElevado))
            }
            temporada.andado?.let { BarraDeAndado(it, Modifier.align(Alignment.BottomStart)) }
        }
        Spacer(Modifier.height(10.dp))
        Text(temporada.rotulo, style = MaterialTheme.typography.titleSmall, color = Cores.texto)
        Text(
            text = buildList {
                add("${temporada.quantos} episódio" + if (temporada.quantos > 1) "s" else "")
                if (temporada.vistos > 0) {
                    add("${temporada.vistos} visto" + if (temporada.vistos > 1) "s" else "")
                }
            }.joinToString("  ·  "),
            style = Tipo.rotulo,
            color = Cores.textoApagado,
        )
    }
}

/// `2021 · 2 temporadas · 18 episódios · 3 vistos` — cada pedaço só entra se
/// tiver o que dizer (§24).
internal fun contagem(temporadas: Int, episodios: Int, vistos: Int, ano: Int?): String = buildList {
    ano?.let { add(it.toString()) }
    if (temporadas > 0) add("$temporadas temporada" + if (temporadas > 1) "s" else "")
    if (episodios > 0) add("$episodios episódio" + if (episodios > 1) "s" else "")
    if (vistos > 0) add("$vistos visto" + if (vistos > 1) "s" else "")
}.joinToString("  ·  ")

@Composable
internal fun BarraDeAndado(fracao: Float, modifier: Modifier = Modifier) {
    Box(modifier.fillMaxWidth().height(4.dp).background(Color.Black.copy(alpha = 0.55f))) {
        Box(
            Modifier
                .fillMaxWidth(fracao.coerceIn(0f, 1f))
                .height(4.dp)
                .background(Cores.destaque),
        )
    }
}

@Composable
internal fun MarcaDeVisto(modifier: Modifier = Modifier) {
    Box(
        modifier.size(26.dp).clip(CircleShape).background(Color.Black.copy(alpha = 0.78f)),
        contentAlignment = Alignment.Center,
    ) {
        Text("✓", style = Tipo.rotulo, color = Cores.destaque)
    }
}

@Composable
private fun EsqueletoDaSerie(modifier: Modifier = Modifier) {
    /// §15: quadros vazios, e nunca a palavra «carregando».
    Column(modifier.fillMaxSize().background(Cores.fundo)) {
        Box(Modifier.fillMaxWidth().aspectRatio(16f / 9f).background(Cores.fundoElevado))
        Spacer(Modifier.height(20.dp))
        Row(
            Modifier.padding(horizontal = 20.dp),
            horizontalArrangement = Arrangement.spacedBy(14.dp),
        ) {
            repeat(2) {
                Box(
                    Modifier
                        .width(230.dp)
                        .aspectRatio(16f / 9f)
                        .clip(RoundedCornerShape(12.dp))
                        .background(Cores.fundoElevado),
                )
            }
        }
    }
}

File: ios/Odeon/Telas/TelaDaSerie.swift
import SwiftUI

/// A ficha de uma série — o desenho aprovado em 18/08/2026, no iPhone.
///
/// ## ⚠️ Ela é a **primeira** tela de série deste app
///
/// Até hoje o iOS não tratava série de forma alguma: `item.eSerie` mudava um
/// rótulo no cartaz e mais nada, e tocar numa série abria a ficha da obra — que
/// dá **404**, porque o id de uma série é de coleção, não de obra. O mesmo
/// defeito que a TV tinha e que a locadora já havia pago.
///
/// O desenho é o da TV e o do Android, nas mesmas palavras e na mesma ordem:
/// onde eu parei, o que é isto, onde está o resto. Ver `docs/SERIES.md` no
/// `android/`.
struct TelaDaSerie: View {
    let odeon: RepositorioOdeon
    @State var modelo: ModeloDaSerie
    let aoTocar: (ObraDaLista, Double) -> Void
    let aoFechar: () -> Void

    /// ⚠️ Guarda o **número**, e não a `TemporadaDaSerie`. O destino precisa de
    /// `Hashable`, e uma temporada carrega os episódios dentro — torná-la
    /// `Hashable` faria a navegação comparar listas inteiras a cada quadro. O
    /// número identifica, e a tela de lá relê do mesmo modelo.
    @State private var temporadaAberta: Int?

    var body: some View {
        ZStack {
            Cores.fundo.ignoresSafeArea()

            if let erro = modelo.erro {
                Recado(titulo: "a série não abriu", detalhe: erro) {
                    Button("tentar de novo") { Task { await modelo.carregar() } }
                        .foregroundStyle(Cores.destaque)
                    Button("voltar", action: aoFechar).foregroundStyle(Cores.textoApagado)
                }
            } else if modelo.carregando {
                Esqueleto()
            } else if modelo.vazio {
                Recado(
                    titulo: "esta série está sem episódios",
                    detalhe: "o acervo tem a série, mas nenhum arquivo casou com ela",
                ) {
                    Button("voltar", action: aoFechar).foregroundStyle(Cores.destaque)
                }
            } else {
                conteudo
            }
        }
        .task { if modelo.carregando { await modelo.carregar() } }
        .navigationDestination(item: $temporadaAberta) { numero in
            TelaDaTemporada(
                odeon: odeon,
                modelo: modelo,
                numero: numero,
                aoTocar: { ep in aoTocar(ep, ep.ondeParou ?? 0) },
            )
        }
    }

    private var conteudo: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 0) {
                /// ⚠️ Bloco 16:9 no topo, e não a tela inteira: numa coluna de
                /// celular, arte atrás de texto vira textura.
                ZStack(alignment: .bottom) {
                    Color.clear
                        .aspectRatio(16.0 / 9.0, contentMode: .fit)
                        .background {
                            ArteDoOdeon(odeon: odeon, caminho: modelo.panoDeFundo)
                        }
                        .clipped()
                    /// O véu pro letreiro encostar na arte sem flutuar sobre ela.
                    LinearGradient(
                        colors: [.clear, Cores.fundo],
                        startPoint: .init(x: 0.5, y: 0.35),
                        endPoint: .bottom,
                    )
                    .allowsHitTesting(false)
                }

                VStack(alignment: .leading, spacing: 0) {
                    Text(modelo.titulo)
                        .font(Tipo.letreiro(30))
                        .foregroundStyle(Cores.texto)
                    Text(contagem)
                        .font(Tipo.rotulo(11))
                        .tracking(Tipo.espacoDoRotulo)
                        .foregroundStyle(Cores.textoApagado)
                        .padding(.top, 8)

                    /// ⚠️ A sinopse da série — 115 das 120 têm. Quem não tem não
                    /// ganha parágrafo nenhum (§24).
                    if let sinopse = modelo.sinopse, !sinopse.isEmpty {
                        Text(sinopse)
                            .font(.subheadline)
                            .foregroundStyle(Cores.textoApagado)
                            .lineLimit(4)
                            .padding(.top, 12)
                    }

                    if let onde = modelo.ondeParou {
                        botao(onde).padding(.top, 20)
                        /// §24: «do começo» só havendo meio.
                        if onde.comecado {
                            Button("do começo") { aoTocar(onde.episodio, 0) }
                                .font(.subheadline)
                                .foregroundStyle(Cores.destaque)
                                .frame(minHeight: 44)
                        }
                    }

                    RotuloDeSecao(texto: "temporadas", contagem: "\(modelo.temporadas.count)")
                        .padding(.top, 22)
                }
                .padding(.horizontal, 20)

                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(alignment: .top, spacing: 14) {
                        ForEach(modelo.temporadas) { t in
                            CartaoDaTemporada(odeon: odeon, temporada: t) {
                                temporadaAberta = t.numero
                            }
                        }
                    }
                    .padding(.horizontal, 20)
                }
                .padding(.bottom, 28)
            }
        }
        .background(Cores.fundo)
    }

    private func botao(_ onde: OndeParouNaSerie) -> some View {
        let cod = onde.episodio.codigo.map { "\($0) · " } ?? ""
        return Button {
            aoTocar(onde.episodio, onde.comecado ? (onde.episodio.ondeParou ?? 0) : 0)
        } label: {
            Text((onde.comecado ? "▸  continuar  " : "▸  começar  ") + cod + onde.episodio.title)
                .font(.headline)
                .foregroundStyle(Cores.fundo)
                .lineLimit(1)
                .frame(maxWidth: .infinity, minHeight: 52)
                .background(Cores.destaqueQuente, in: .capsule)
        }
        .buttonStyle(.plain)
    }

    /// `2021 · 2 temporadas · 18 episódios · 3 vistos` — cada pedaço só entra se
    /// tiver o que dizer (§24).
    private var contagem: String {
        var partes: [String] = []
        if let ano = modelo.ano { partes.append("\(ano)") }
        let t = modelo.temporadas.count
        if t > 0 { partes.append("\(t) temporada" + (t > 1 ? "s" : "")) }
        let e = modelo.quantosEpisodios
        if e > 0 { partes.append("\(e) episódio" + (e > 1 ? "s" : "")) }
        let v = modelo.quantosVistos
        if v > 0 { partes.append("\(v) visto" + (v > 1 ? "s" : "")) }
        return partes.joined(separator: "  ·  ")
    }
}

private struct CartaoDaTemporada: View {
    let odeon: RepositorioOdeon
    let temporada: TemporadaDaSerie
    let aoTocar: () -> Void

    var body: some View {
        Button(action: aoTocar) {
            VStack(alignment: .leading, spacing: 8) {
                /// ⚠️ `Color.clear` com `aspectRatio` e a arte no `background` —
                /// e **não** a imagem com `scaledToFill`. Uma imagem que preenche
                /// reivindica a largura intrínseca dela, e as temporadas sairiam
                /// com larguras diferentes. É o defeito que este app já pagou
                /// cinco vezes; ver o `MenuDeDVD`.
                /// ## ⚠️ O cartão virou **retrato** — 18/08/2026
                ///
                /// Ele era 16:9 porque a única imagem era o `still` do primeiro
                /// episódio. O servidor passou a mandar o **pôster da temporada**
                /// do TMDB, e pôster é 2:3 — num quadro deitado a `Temporada 1`
                /// do Arcane virava um olho. **A moldura segue a imagem.**
                ZStack(alignment: .bottomLeading) {
                    Color.clear
                        .aspectRatio(2.0 / 3.0, contentMode: .fit)
                        .background { ArteDoOdeon(odeon: odeon, caminho: temporada.arte) }
                        .background(Cores.fundoElevado)
                        .clipShape(.rect(cornerRadius: 12))
                    if let andado = temporada.andado {
                        BarraDeAndado(fracao: andado)
                    }
                }
                Text(temporada.rotulo)
                    .font(.subheadline.weight(.semibold))
                    .foregroundStyle(Cores.texto)
                Text(sub)
                    .font(Tipo.rotulo(11))
                    .tracking(Tipo.espacoDoRotulo)
                    .foregroundStyle(Cores.textoApagado)
            }
            .frame(width: 150)
        }
        .buttonStyle(.plain)
    }

    private var sub: String {
        var partes = ["\(temporada.quantos) episódio" + (temporada.quantos > 1 ? "s" : "")]
        if temporada.vistos > 0 {
            partes.append("\(temporada.vistos) visto" + (temporada.vistos > 1 ? "s" : ""))
        }
        return partes.joined(separator: "  ·  ")
    }
}

struct BarraDeAndado: View {
    let fracao: Double

    var body: some View {
        GeometryReader { g in
            ZStack(alignment: .leading) {
                Color.black.opacity(0.55)
                Cores.destaque.frame(width: g.size.width * min(max(fracao, 0), 1))
            }
        }
        .frame(height: 4)
    }
}

struct MarcaDeVisto: View {
    var body: some View {
        Image(systemName: "checkmark")
            .font(.system(size: 11, weight: .bold))
            .foregroundStyle(Cores.destaque)
            .frame(width: 24, height: 24)
            .background(.black.opacity(0.78), in: .circle)
    }
}

private struct Esqueleto: View {
    /// §15: quadros vazios, e nunca a palavra «carregando».
    var body: some View {
        VStack(alignment: .leading, spacing: 20) {
            Color.clear.aspectRatio(16.0 / 9.0, contentMode: .fit)
                .background(Cores.fundoElevado)
            HStack(spacing: 14) {
                ForEach(0..<2, id: \.self) { _ in
                    Color.clear
                        .frame(width: 230)
                        .aspectRatio(16.0 / 9.0, contentMode: .fit)
                        .background(Cores.fundoElevado)
                        .clipShape(.rect(cornerRadius: 12))
                }
            }
            .padding(.horizontal, 20)
            Spacer()
        }
    }
}

struct Recado<Acoes: View>: View {
    let titulo: String
    var detalhe: String?
    @ViewBuilder let acoes: () -> Acoes

    var body: some View {
        VStack(alignment: .leading, spacing: 10) {
            Text(titulo).font(.title3).foregroundStyle(Cores.texto)
            if let detalhe {
                Text(detalhe).font(.subheadline).foregroundStyle(Cores.textoApagado)
            }
            HStack(spacing: 16) { acoes() }.frame(minHeight: 44)
        }
        .padding(28)
    }
}
